tests/CMakeLists.txt

```diff
@@ -7,12 +7,273 @@ if( GPERFTOOLS_FOUND )
     list( APPEND PLATFORM_SPECIFIC_LIBS tcmalloc )
 endif()
 
+INCLUDE(CTest)
+
+ENABLE_TESTING()
+
 file(GLOB UNIT_TESTS "tests/*.cpp")
-add_executable( chain_test ${UNIT_TESTS} )
+
+include(BoostTestTargets.cmake)
+
+function(parse_unit_tests RESULT)
+   set(SOURCES)
+	foreach(_element ${ARGN})
+      list(APPEND SOURCES "${_element}")
+	endforeach()
+
+   set(tests)
+
+   foreach(src ${SOURCES})
+      file(READ ${src} thefile)
+      #string(REGEX MATCH "" test_suite "${thefile}" )
+      string(REGEX MATCH "BOOST_FIXTURE_TEST_SUITE\\([A-Za-z0-9_, ]*\\)" test_suite "${thefile}" )
+
+      if( NOT (test_suite STREQUAL "") )
+         string(SUBSTRING "${test_suite}" 25 -1 test_suite)
+         string(FIND "${test_suite}" "," comma_loc )
+         string(SUBSTRING "${test_suite}" 0 ${comma_loc} test_suite)
+         string(STRIP "${test_suite}" test_suite)
+
+         string( REGEX MATCHALL "BOOST_AUTO_TEST_CASE\\([A-Za-z0-9_, ]*\\)" cases "${thefile}" )
+
+         foreach( test_case ${cases} )
+            string(SUBSTRING "${test_case}" 22 -1 test_case)
+            string(FIND "${test_case}" ")" paren_loc )
+            string(SUBSTRING "${test_case}" 0 ${paren_loc} test_case)
+            string(STRIP "${test_case}" test_case)
+
+            list(APPEND tests "${test_suite}/${test_case}")
+         endforeach()
+      endif()
+   endforeach()
+
+   set(${RESULT} ${tests} PARENT_SCOPE)
+endfunction()
+
+parse_unit_tests(TEST_CASES ${UNIT_TESTS})
+
+#message(STATUS "${TEST_CASES}")
+
+add_boost_test( chain_test
```

Comment:
It looks like work was done to parse unit test suites and names automagically, but they are still listed manually.

Reply:
Using a regex to pull out `BOOST_AUTO_TEST_CASE` gets some tests that have been commented out or in the case of SMTS, not compiled in on certain builds. When a test is ran and not found, it returns a non-zero exit code, which would result in CI failing. Some more fine tuning of the build system needs to be done. But for now, I parsed and logged the tests cases and then manually removed the ones we don't want to run. I am viewing this as a first step, but it accomplishes the goal of reducing CI time.